chain-tx-enclave-next/mls/src/tree_math.rs

```diff
@@ -199,23 +209,52 @@ impl NodeSize {
 
     /// The direct path of a node, ordered from the root
     /// down, including the root
+    ///
+    /// Only root node returns empty vector
     #[inline]
     pub fn direct_path(self, leafs: LeafSize) -> Vec<NodeSize> {
+        let root = NodeSize::root(leafs);
         let mut d = Vec::new();
-        if let Some(mut p) = self.parent(leafs) {
-            let r = NodeSize::root(leafs);
-            while p != r {
-                d.push(p);
-                // no panic: only root node returns `None`, p is definitly not root node.
-                p = p.parent(leafs).unwrap();
-            }
-            if self != r {
-                d.push(r);
+        let mut p = self;
+        while p != root {
+            // no panic: only root node returns `None`, p is not root node because of the while
+            // condition.
+            p = p.parent(leafs).unwrap();
+            d.push(p);
+        }
+        d
+    }
+
+    /// The other child of the node's parent.  Root's sibling is itself.
+    ///
+    /// Only root node returns None
+    pub fn sibling(self, leafs: LeafSize) -> Option<NodeSize> {
+        if let Some(p) = self.parent(leafs) {
+            if self < p {
+                p.right(leafs)
+            } else {
+                // impossible they are equal
+                assert!(self > p);
+                p.left()
             }
-            d
         } else {
-            vec![]
+            None
+        }
+    }
+
+    /// The copath of the node is the siblings of the nodes on its direct
+    /// path (including the node itself)
+    pub fn copath(self, leafs: LeafSize) -> Vec<NodeSize> {
```

Comment:
I guess these are not yet used in the code? but they'll be necessary for populating the paths -- perhaps add the tests with precomputed values

Reply:
done